WordPress/gutenberg
# Themes: Support title in templateParts

## Description
Adds "title" support in `templateParts` theme.json configuration.

Fixes #34723.

Cc @gziolo, I saw that you introduced i18n for the `customTemplates` title, and I just wanted to check if I'm handling it correctly here.

## How has this been tested?
1. Update `templateParts` definition in theme, e.g. Blockbase.
2. Go to the Site Editor -> Template Parts.
3. Check that provided "title" is displayed.

## Example
```json
"templateParts": [
      {
	      "name": "header",
	      "title": "Header",
	      "area": "header"
      },
      {
	      "name": "footer",
	      "title": "Footer",
	      "area": "footer"
      }
]
```

## Screenshots 
| Before | After |
| --- | --- |
| ![CleanShot 2021-10-14 at 13 09 52](https://user-images.githubusercontent.com/240569/137288157-fe487be9-8d23-4f1e-91a6-5e1a2677e81c.png) | ![CleanShot 2021-10-14 at 13 10 20](https://user-images.githubusercontent.com/240569/137288165-f0403c47-7e1d-402f-9ad1-ccffb09263aa.png) |

## Types of changes
Enhancement

## Checklist:
